src/Libraries/CoreNodeModels/DropDown.cs

```diff
@@ -173,32 +181,23 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
             if (selectedIndex < 0)
             {
                 Warning(Dynamo.Properties.Resources.NothingIsSelectedWarning);
-                selectedString = String.Empty;
+                selectedString = string.Empty;
             }
             else
             {
-                selectedString = selectedIndex > Items.Count - 1 ? String.Empty : GetSelectedStringFromItem(Items.ElementAt(selectedIndex));
+                selectedString = selectedIndex > Items.Count - 1 ? string.Empty : GetSelectedStringFromItem(Items.ElementAt(selectedIndex));
             }
         }
 
         protected override bool UpdateValueCore(UpdateValueParams updateValueParams)
         {
-            string name = updateValueParams.PropertyName;
             string value = updateValueParams.PropertyValue;
 
-            if (name == "Value" && value != null)
+            if (updateValueParams.PropertyName == "Value" && value != null)
             {
-                selectedIndex = ParseSelectedIndex(value, Items);
-                if (selectedIndex < 0)
-                {
-                    Warning(Dynamo.Properties.Resources.NothingIsSelectedWarning);
-                    selectedString = String.Empty;
```

Comment:
Did I miss anything, the edge case here is skipped? I believe if the parse failed, the index will be negative

Reply:
We're now using the property setter to set SelectedIndex, so that we get the PropertyChanged event. That setter already has an out-of-bounds check. I notice it doesn't have the Warning though, I'll add that.

Reply:
Unless there's a pathway which should be able to set SelectedIndex to -1 without raising a warning. I don't _think_ so...

Reply:
Added the warning back here. The rest of the edge case check is handled in the setter.

Reply:
Thanks, should be good then